src/app/frank/fd_frank_verify.c

```diff
@@ -171,15 +264,158 @@ run( fd_frank_args_t * args ) {
+        SEEMS UNLKELY AS THEY WOULD EITHER BE BEHIND THE INBOUND OR BE
+        A MITM THAT COULD JUST DISCARD INBOUND TRAFFIC).
+
+        When running synthetic load though, we only have a very limited
+        set of messages and this dedup will be overly aggressive (as it
+        will spuriously matching earlier synthetic packets since they
+        are not resigned continuously)  So we just mock this up for the
+        time being. */
+
+    ulong ha_tag = *sig;
+    int ha_dup;
+    FD_TCACHE_INSERT( ha_dup, tcache_oldest, _tcache_ring, tcache_depth, _tcache_map, tcache_map_cnt, ha_tag );
+    if( FD_UNLIKELY( ha_dup ) ) { /* optimize for the non dup case */
+      accum_ha_filt_cnt++;
+      accum_ha_filt_sz += payload_sz; //WW accum_ha_filt_sz += msg_framing + msg_sz;
+      now = fd_tickcount();
+      continue;
+    }
+
+    /* We appear to have a message to verify.  So verify it.
+
+       When running synthetic load, the synthetic data will not fail
+       at this point so we fake up some configurable rate of errors to
+       stress out the monitoring.  (We could also slightly more
+       expensively get the same effect by corrupting the udp_payload
+       region before the verify.) */
+
+    int err = fd_ed25519_verify( msg, msg_sz, sig, public_key, sha );
+    if (err) {
+      sigvfy_fail_cnt ++;
+      //FD_LOG_WARNING(( "fd_ed25519_verify failed for mcache[%lu], fail/pass: %lu/%lu",
+      //  vin_seq_found, sigvfy_fail_cnt, sigvfy_pass_cnt ));
+      now = fd_tickcount();
+      continue;
+    }
+    else {
+      sigvfy_pass_cnt ++;
+    }
+
+    /* Packet looks superficially good.  Forward it.  If somebody is
+       opening multiple connections (which would potentially flow
+       steered to different verify tiles) and spammed these
+       connections with the same transaction, ha dedup here is likely
+       to miss that.   But the dedup tile that muxes all the inputs
+       will take care of that.  (The use of QUIC and the like should
+       also strongly reduce the economic incentives for this
+       behavior.)
+
+       When running synthetic load, we have the same problem we had
+       above.  So we use a signature that will match with the desired
+       probability. */
+
+    /* Note that sig is now guaranteed to be not FD_TCACHE_TAG_NULL
+       and we use the least significant 64-bits of the SHA-512 hash
+       for dedup purposes. */
+
     now = fd_tickcount();
+    ulong tspub = fd_frag_meta_ts_comp( now );
+    int   ctl_som  = 1;
+    int   ctl_eom  = 1;
+    ulong   ctl    = fd_frag_meta_ctl( 0, ctl_som, ctl_eom, 0 );
+    ulong   tsorig = tspub;
+    fd_mcache_publish( mcache, depth, seq, ha_tag, chunk, vin_data_sz, ctl, tsorig, tspub );
+
+    chunk = fd_dcache_compact_next( chunk, vin_data_sz, chunk0, wmark );
+    seq   = fd_seq_inc( seq, 1UL );
+    cr_avail--;
+
+    if( FD_UNLIKELY( !ctl_eom ) ) ctl_som = 0;
+    else {
+      ctl_som = 1;
+    }
   }
+
+  /* Clean up */
```

Comment:
This should all be removed, there is no cleanup in the current shutdown model (and these would all get BADSYS anyway).